subgraphs/rocket-pool/src/entities/token.ts

```diff
@@ -0,0 +1,153 @@
+
+  if (!token) {
+    token = new Token(tokenId);
+
+    if (tokenAddress == Address.fromString(ETH_ADDRESS)) {
+      token.name = ETH_NAME;
+      token.symbol = ETH_SYMBOL;
+      token.decimals = 18;
+    } else {
+      token.name = fetchTokenName(tokenAddress);
+      token.symbol = fetchTokenSymbol(tokenAddress);
+      token.decimals = fetchTokenDecimals(tokenAddress) as i32;
+    }
+  }
+
+  // Optional lastPriceUSD and lastPriceBlockNumber, but used in financialMetrics
+  // if (tokenAddress.equals(Address.fromString(RETH_ADDRESS))) {
+  //   let rethpriceUSD: BigDecimal | null = null;
+  //   rethpriceUSD = getRETHpriceUSD(
+  //     Address.fromString(RETH_ADDRESS),
+  //     blockNumber
+  //   );
+  //   log.warning("[getOrCreateToken]rethPriceUSD call result: {} ", [
+  //     rethpriceUSD.toString(),
+  //   ]);
+
+  //   token.lastPriceUSD = rethpriceUSD;
+  //   token.lastPriceBlockNumber = blockNumber;
+  //   token.save();
+
+  //   return token;
+  // }
+  if (
+    token.lastPriceBlockNumber &&
+    token.lastPriceBlockNumber!.equals(blockNumber)
+  ) {
+    token.save();
+    return token;
+  }
+  let price = getUsdPricePerToken(tokenAddress);
+
+  if (price.reverted) {
+    token.lastPriceUSD = BIGDECIMAL_ZERO;
+  } else {
+    token.lastPriceUSD = price.usdPrice.div(price.decimalsBaseTen);
+  }
+
+  token.lastPriceBlockNumber = blockNumber;
+  token.save();
+
+  return token;
+}
+
+function fetchTokenName(tokenAddress: Address): string {
+  const tokenContract = ERC20.bind(tokenAddress);
+  const call = tokenContract.try_name();
+  if (call.reverted) {
+    return tokenAddress.toHexString();
+  } else {
+    return call.value;
+  }
+}
+
+function fetchTokenSymbol(tokenAddress: Address): string {
+  const tokenContract = ERC20.bind(tokenAddress);
+  const call = tokenContract.try_symbol();
+  if (call.reverted) {
+    return " ";
+  } else {
+    return call.value;
+  }
+}
+
+function fetchTokenDecimals(tokenAddress: Address): number {
+  const tokenContract = ERC20.bind(tokenAddress);
+  const call = tokenContract.try_decimals();
+  if (call.reverted) {
+    return 0;
+  } else {
+    return call.value;
+  }
+}
+
+function getRETHpriceUSD(
+  tokenAddress: Address,
+  blockNumber: BigInt
+): BigDecimal {
+  const RETHContract = RETH.bind(tokenAddress);
+
+  const rethCall = RETHContract.try_getExchangeRate();
+  let RETHratio = BIGDECIMAL_ZERO;
+  let RETHpriceUSD = BIGDECIMAL_ZERO;
+  if (rethCall.reverted) {
+    // log.error("[getRETHpriceUSD] rethCall Reverted", []);
+  } else {
+    // log.error("[getRETHpriceUSD] rethCall resolved to : {}", [
+    //   rethCall.value.toString(),
+    // ]);
+    RETHratio = bigIntToBigDecimal(rethCall.value.div(ONE_ETH_IN_WEI));
+
+    if (RETHratio.notEqual(BIGDECIMAL_ZERO)) {
+      let ETH = getOrCreateToken(Address.fromString(ETH_ADDRESS), blockNumber);
+      let ethPriceUSD = ETH.lastPriceUSD;
+
+      if (ethPriceUSD) {
+        RETHpriceUSD = ethPriceUSD.div(RETHratio);
```

Comment:
And this then would be the other way around: `RETHPriceUSD = ethPriceUSD.times(RETHRatio)`.